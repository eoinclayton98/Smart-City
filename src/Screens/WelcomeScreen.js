import React, { Component } from 'react';
import { View, Text, StyleSheet, Button, Image } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';


import Footer from '../Components/Footer';

/*
  Welcome Screen:
  Initial screen displayed as the app is loaded.
  Displays a logo and short description about the app.
 */

export default class WelcomeScreen extends Component {


    static navigationOptions = { // Setting the heading of the screen
        title: 'Smart-City',
        headerMode: 'float',
        headerTitleAlign: 'center',
        headerStyle: {
            backgroundColor: '#07EFEF',

        },

    };


    render() {
        return (
            <View style={styles.screen}>
                <ScrollView>{/*enables the content to be scrollable*/}
                    
                    <View style={{ width: '100%', height: '40%' }}>
                        <Image source={require('../assets/logo.png')}
                                style={{ width: 220, height: 200, alignSelf : 'center' }}
                                />
                    </View>
                <View style={styles.text}>
                    <Text style={styles.heading}>Hello, Welcome to Smart-City</Text>
                    <Text>This app allows you to take a photo of a building or landmark and our system will return information relevant to this building.</Text>
                    <Text>You can also chose between several predesigned routes to use as a tour guide or see what buildings and landmarks are nearby.</Text>

                        
                        <View style={styles.button}> 
                            <Button title="Continue" onPress={() => this.props.navigation.navigate('Home')} />{/*Button to navigate to the home screen*/}
                        </View>


                        
                </View>

                </ScrollView>

                <Footer>


                </Footer>
 
            </View>
        );
    }
}



 
const styles = StyleSheet.create({
    screen: {
        flex: 1,
        padding: '5%',
        paddingTop: '2%',
        alignItems: 'center',
        backgroundColor: 'white',
    },
    text: {
        top: 25,
        alignItems: 'center',
        justifyContent: 'center',
    },
    heading: {
        fontSize: 18,
        marginTop:'2%',
        marginBottom: '2%',
        fontWeight: 'bold',
        width: '95%',
        textAlign: 'center',
    },
    button: {
        width: '50%',
        marginTop: '3%',
        marginBottom: '10%',
    }
})

