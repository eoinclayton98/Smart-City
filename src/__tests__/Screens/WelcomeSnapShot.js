import 'react-native';
import React from 'react';
import WelcomeScreen from '../../Screens/WelcomeScreen';

import renderer from 'react-test-renderer';

test('Welcome SnapShot', () => {

    const snap = renderer.create(<WelcomeScreen />).toJSON();
    expect(snap).toMatchSnapshot();

});


