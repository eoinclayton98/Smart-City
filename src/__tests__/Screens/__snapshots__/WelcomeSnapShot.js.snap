// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome SnapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "padding": "5%",
      "paddingTop": "2%",
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "height": "40%",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 200,
              "width": 220,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "top": 25,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": "2%",
              "marginTop": "2%",
              "textAlign": "center",
              "width": "95%",
            }
          }
        >
          Hello, Welcome to Smart-City
        </Text>
        <Text>
          This app allows you to take a photo of a building or landmark and our system will return information relevant to this building.
        </Text>
        <Text>
          You can also chose between several predesigned routes to use as a tour guide or see what buildings and landmarks are nearby.
        </Text>
        <View
          style={
            Object {
              "marginBottom": "10%",
              "marginTop": "3%",
              "width": "50%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Continue
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#07EFEF",
        "bottom": 0,
        "height": "10%",
        "position": "absolute",
        "width": 750,
      }
    }
  />
</View>
`;
